wayland-server/src/rust_imp/resources.rs

```diff
@@ -69,20 +70,25 @@ impl ResourceInner {
 
     pub(crate) fn send<I: Interface>(&self, msg: I::Event) {
         if let Some(ref mut conn_lock) = *self.client.data.lock().unwrap() {
-            if !self.is_alive() {
+            let is_alive = self.is_alive();
+
+            if !is_alive {
                 return;
             }
+
             let destructor = msg.is_destructor();
             let msg = msg.into_raw(self.id);
-            if ::std::env::var_os("WAYLAND_DEBUG").is_some() {
-                eprintln!(
-                    " -> {}@{}: {} {:?}",
+
+            if WAYLAND_DEBUG.load(Ordering::Relaxed) {
+                debug::print_send_event(
                     I::NAME,
                     self.id,
+                    is_alive,
                     self.object.events[msg.opcode as usize].name,
-                    msg.args
+                    &msg.args,
                 );
             }
```

Comment:
The client lib has a notion of `zombie` should this one be changed as well? What it's your thoughts on how to handle early return.